### DISCUSSION TITLE
versioned-messages: how to use?

### DISCUSSION THREAD

**Discussion by stephan-buckmaster** [AllYourBot/hostedgpt#276] [General]
I checked out the versioned-messages branch. I'm not sure how to use it. In fact, I just noticed the "Edit" link also appears under posted messages in the main branch and it doesn't work for me. I'm using Chrome, nothing is logged in browser console or server side. I am just running "rails server," not docker, so could I be missing something? How to track this down?

**Comment by keithschacht** [comment]
@stephan-buckmaster I have another PR I'm just about done with, I should have it tee'd up later today. Then I'll merge in the existing PR (which is just the backend) and the new PR which will be the front-end for it. 🤞 that I got this right. It's turned out to be a more complex feature to implement than I expected but I managed to figure out how to avoid creating a whole graph data structure for it, which would have made it even more of a pain. :)

**Comment by keithschacht** [comment]
@stephan-buckmaster I finally finished this up and merged it in. I would love for you to kick the tires and see if you run into any issues with it!

And now I can spend some more time looking at your PR, I've been a bit distracted trying to hold all the context of this PR in my head as I tried to land this. It turned out to be quite a bit more complex than I anticipated. But in the end I'm pleased with the implementation and the user experience of it.
